pkg/appendable/io.go

```diff
@@ -109,28 +121,49 @@ func ReadIndexFile(r io.Reader, data DataHandler) (*IndexFile, error) {
 
 		// read the data ranges
 		f.EndByteOffsets = make([]uint64, ifh.DataCount)
-		f.Checksums = make([]uint64, ifh.DataCount)
 		for i := 0; i < int(ifh.DataCount); i++ {
 			if f.EndByteOffsets[i], err = encoding.ReadUint64(r); err != nil {
 				return nil, fmt.Errorf("failed to read data range: %w", err)
 			}
 		}
-		start := uint64(0)
+
+		// read the checksums
+		f.Checksums = make([]uint64, ifh.DataCount)
 		for i := 0; i < int(ifh.DataCount); i++ {
 			if f.Checksums[i], err = encoding.ReadUint64(r); err != nil {
-				return nil, fmt.Errorf("failed to read data range: %w", err)
+				return nil, fmt.Errorf("failed to read checksum: %w", err)
+			}
+		}
+
+		fmt.Printf("All checksums: %v\n", f.Checksums)
+
+		startIndex := 0
+		start := uint64(0)
+		if _, isCSV := data.(CSVHandler); isCSV && len(f.EndByteOffsets) > 0 {
+			start = f.EndByteOffsets[0]
+			startIndex = 1
+		}
+		for i := startIndex; i < int(ifh.DataCount); i++ {
+
+			fmt.Printf("Current start idx: %d, position: %d\n", i, start)
+
+			if _, isCsv := data.(CSVHandler); isCsv {
+				if i > 1 {
+					start -= 1
```

Comment:
this is a hotfix solution that worked.

The reason was the start position was +1 from the ideal start byte offset